### DISCUSSION TITLE
Octo4a Installation Issue: "invalid GMT offset" Error Prevents OctoPrint from Starting

### DISCUSSION THREAD

**Discussion by javi-1111** [feelfreelinux/octo4a#508] [Q&A]
Hi everyone,

I recently installed Octo4a on my Android device, but I'm encountering an issue where the OctoPrint server fails to start. The installation process seems to go smoothly, but the final step throws a RuntimeError: invalid GMT offset, and OctoPrint doesn't launch.

Here is the relevant section from the installation log:

💻No bootstrap detected, proceeding with installation
💻 Arch: arm
💻 Downloading bootstrap f4f81ac1 from https://github.com/feelfreelinux/octo4a-bootstrap-builder/releases/download/vf4f81ac1-1.10.2/bootstrap-1.10.2-armv7a.zip
💻 Request to https://objects.githubusercontent.com/github-production-release-asset-2e65be/795680395/2d9234a9-d8f2-4cbc-a0df-a2be327d171e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240830%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240830T184521Z&X-Amz-Expires=300&X-Amz-Signature=3794d00e4feb8a2566e56d2ccc5ad99881201c481c3bfd850d2f20be1a7ccb9d&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=795680395&response-content-disposition=attachment%3B%20filename%3Dbootstrap-1.10.2-armv7a.zip&response-content-type=application%2Foctet-stream returned status code 200
💻 Bootstrap downloaded, extracting it...
💻 >ls
💻 /storage/emulated/0/Android/data/com.octo4a/files
🐸 bin
🐸 bin-pre5
🐸 build-version.txt
🐸 entrypoint.sh
🐸 fake_proc_stat
🐸 libexec
🐸 libexec-pre5
🐸 rootfs.tar.xz
💻 >ls
💻 /storage/emulated/0/Android/data/com.octo4a/files
🐸 No bootstrap detected, extracting
💻 Bootstrap extracted, setting it up...
💻 >chmod -R 755 /mnt/external/
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 >cp -rf /home/octoprint/extensions /mnt/external/
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 >chown root:root /mnt/external/
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 >mkdir -p /mnt/external/.octoprint/plugins
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 >cp /home/octoprint/comm-fix.py /mnt/external/.octoprint/plugins
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 Bootstrap installed
💻 >ls /home/octoprint/octoprint-venv/bin/octoprint && echo "available"
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 >LD_PRELOAD=/home/octoprint/ioctl-hook.so /home/octoprint/octoprint-venv/bin/octoprint serve -b /mnt/external/.octoprint --iknowwhatimdoing
💻 /storage/emulated/0/Android/data/com.octo4a/files
💻 Dependencies installed
🐙 Traceback (most recent call last):
🐙   File "/home/octoprint/octoprint-venv/bin/octoprint", line 5, in <module>
🐙     from octoprint import main
🐙   File "/home/octoprint/octoprint-venv/lib/python3.11/site-packages/octoprint/__init__.py", line 3, in <module>
🐙     import logging as log
🐙   File "/usr/lib/python3.11/logging/__init__.py", line 26, in <module>
🐙     import sys, os, time, io, re, traceback, warnings, weakref, collections.abc
🐙 RuntimeError: invalid GMT offset
I've already checked my device's date and time settings and ensured that they are set to automatic. I've also tried setting the time zone manually, but the problem persists.

Could this be an issue with how Octo4a handles the time zone settings, or is there something else I'm missing? Any help or suggestions would be greatly appreciated!

Thank you in advance!

**Comment by ResRobert** [comment]
Have you found a solution? Try to install today and get the same error. Tried older version and latest but its the same. I even try to set the time zone manually on phone but did not help.

Have anyone found a solution or what is causing the error?

**Comment by megacongrio78** [comment]
Have u found any solution??
Same problem here, have tried everything, nothing seems to work. It's frustrating 
I even tried to change manually 1 by 1 gmt, with no luck.
I'm trying to install in samsung a13 os 14.

**Reply by ResRobert** [reply]
nope - I guess its dead - purchase a Pi and set it up there.. it is a nice solution to have it on old android phone but without support and working state its worthless

**Reply by megacongrio78** [reply]
They all switch to expensive printers I guess...
Pitty, when I saw this solution it blows my mind, it's been a week that I'm looking for a solution...
Thanks for the reply!

**Reply by megacongrio78** [reply]
I manage to solve, switching to a older phone with android 12...

**Reply by ResRobert** [reply]
Thats great to know, it opens a lot of other issue when android 12 gets unsupported, but maybe they will make support for A14 until rhen, tnx for feedback
